packages/maskbook/src/social-network-provider/twitter.com/ui/injectPostImageRevealer.tsx

```diff
@@ -0,0 +1,68 @@
+import { Component, MouseEvent } from 'React'
+import { querySelectorAll } from '../utils/selector'
+import { renderInShadowRoot } from '../../../utils/shadow-root/renderInShadowRoot'
+import { DOMProxy } from '@dimensiondev/holoflows-kit'
+import { Flags } from '../../../utils/flags'
+
+let buttonUrls = new Set<String>()
+
+class RevealEncryptedButton extends Component<{ elem: HTMLElement }, { isHidden: boolean, buttonText: string }> {
+    constructor(props: { elem: HTMLElement }) {
+        super(props)
+        this.state = { isHidden: false, buttonText: 'SHOW ENCRYPTED IMAGE' }
+        this.toggleElement()
+    }
+
+    toggleElement = (): void => {
+        const newHidden = !this.state.isHidden
+        this.setState({ isHidden: newHidden })
+        if (newHidden) {
+            // hide
+            this.props.elem.style.height = '0px'
+            this.setState({ buttonText: 'SHOW ENCRYPTED IMAGE' })
+        } else {
+            // show
+            this.props.elem.style.height = ''
+            this.setState({ buttonText: 'HIDE ENCRYPTED IMAGE' })
+        }
+    }
+
+    render() {
+        return (
+            <button
+                style={{ width: '100%', backgroundColor: "transparent", color: "rgba(29,161,242,1.00)" }}
+                onClick={this.toggleElement}>
+                {this.state.buttonText}
+            </button>
+        )
+    }
+}
+
+function getNParent(element: HTMLElement, n: number) {
+    let p: HTMLElement = element
+    for (let i = 0; i < n; i++) {
+        if (!p.parentElement) return null
+        p = p.parentElement
+    }
+    return p
+}
+
+export function injectPostImageRevealerAtTwitter(encryptedUrl: string) {
+    // make sure button only gets injected once per image
+    const stripped = encryptedUrl.split('?')[0]
+    if (buttonUrls.has(stripped)) return
+    buttonUrls.add(stripped)
+
+    // create button
+    const res = querySelectorAll(`img[src*="${stripped}"]`)
```

Comment:
This way only works for images currently on the page. Please compose it with `new MutationObserverWatcher` to watch future images.

Reply:
smart!